firestore.rules

```diff
@@ -0,0 +1,82 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+
+  	function isAdmin(env) {
+      // Check Admin Privileges.
+      return get(/databases/$(database)/documents/env/$(env)/admins/$(request.auth.uid)).data.secret == "!_xEqyCw3Y.6*wrW32EkCz6GUUn7";
+    }
+
+    match /env/{env}/{document=**} {
+    	// Default document rules.
+    	allow read: if request.auth.uid != null;
+      allow write: if false;
+    }
+
+    match /env/{env} { // PRODUCTION or DEVELOPMENT
+
+    	match /admins/{admin} {
+      	allow read, write: if false;
+      }
+
+    	match /children/{child} {
+      	allow write: if request.auth.uid != null;
+      }
+
+      match /classes/{class} {
+      	allow write: if get(resource.data.teacher).id == request.auth.uid || isAdmin(env);
+        allow create, update: if request.auth.uid != null;
+        allow read: if true;
+      }
+
+      match /promos/{promo} {
+      	allow write: if get(resource.data.teacher).id == request.auth.uid || isAdmin(env);
+        allow create, update: if request.auth.uid != null;
+        allow read: if true;
+      }
+
+      // Account Types
+      match /organizations/{organization} {
+      	allow create: if request.auth.uid != null;
+        allow write: if request.auth.uid == resource.id || isAdmin(env);
+      }
+
+      match /parents/{parent} {
+      	allow create: if request.auth.uid != null;
+        allow write: if request.auth.uid == resource.id || isAdmin(env);
+      }
+
+      match /teachers/{teacher} {
+      	allow create: if request.auth.uid != null;
+        allow write: if request.auth.uid == resource.id || isAdmin(env);
```

Comment:
As of right now there is no functionality that utilizes the admin permissions in these rules outside of `teacher` and `organization` account approval. Because of this, we should remove all instances of admin permissions that don't pertain to those specific features.

Also, the account approval should be restricted to `update` permissions only instead of the generous `write` permissions they currently have.

Reply:
Resolved in commit [1c88041](https://github.com/codecontest-org/app/pull/3/commits/1c880416cda627e75968db8cb9f5eb78e622eb55).